### PR TITLE
[TT-14791] log error on missing vault path in api definition

### PR DESCRIPTION
**Pull request by chrisanderton**
Fix gateway panic when API definition contains vault:// reference and vault path does not exist

## Description

When an API definition (Classic or OAS) contains a vault:// reference in any string field, the gateway attempts to load secrets from a hardcoded vault path (secret/data/tyk-apis) during API load. If this path does not exist in Vault, the gateway crashes with a nil pointer dereference panic. 

## Related Issue

https://tyktech.atlassian.net/browse/TT-14791

## Motivation and Context

* Add a nil check for the vault secret response in replaceVaultSecrets() before accessing secret.Data
* Return a descriptive error message when the vault path does not exist
* The existing error handling in replaceSecrets() will log this error and continue, preventing the panic

## How This Has Been Tested

Test cases in jira

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Refactoring or add test (improvements in base code or adds test coverage to functionality)

## Checklist

- [ ] I ensured that the documentation is up to date
- [ ] I explained why this PR updates go.mod in detail with reasoning why it's required
- [ ] I would like a code coverage CI quality gate exception and have explained why

### Ticket Details

<details>
<summary>
<a href="https://tyktech.atlassian.net/browse/TT-14791" title="TT-14791" target="_blank">TT-14791</a>
</summary>

|         |    |
|---------|----|
| Status  | In Code Review |
| Summary | [Innersource] Gateway panics when API definition contains vault:// reference and vault path does not exist |

Generated at: 2026-02-02 08:29:24

</details>

### CODE REVIEW

**Comment by chrisanderton** [comment]
Not easy to add test case for this new nil check without broader refactoring. This ticket is a targetted nil check bug fix, the change is low risk vs. the risk introduced by a refactor to add test coverage for 3 lines of code. Test coverage for vault should be a factor when kv interfaces are consolidated (a different piece of work to this bug fix).

**Comment by chrisanderton** [comment]
@pvormste how about just doing the read path?

In storage/kv/vault.go:
```
type SecretReader interface {
    ReadSecret(path string) (*vaultapi.Secret, error)
}

func (v *Vault) ReadSecret(path string) (*vaultapi.Secret, error) {
    return v.client.Logical().Read(path)
}
```
In gateway/api_definition.go:
```
vault, ok := a.Gw.vaultKVStore.(kv.SecretReader)
...
secret, err := vault.ReadSecret(vaultSecretPath + prefixKeys)
```

keeps the interface minimal vs. full client

**Comment by chrisanderton** [comment]
Path traversal check is outside the scope of this ticket, we've already ballooned in complexity vs the initial nil check only.

**Comment by chrisanderton** [comment]
Note from/to visor - it initially suggested a path traversal risk, then disagreed with the todo. Comment has been removed.

> The TODO comment suggests a path traversal vulnerability exists, but the implementation in `replaceVaultSecrets` does not seem to support this. The Vault path is hardcoded to `secret/data/tyk-apis`, and keys from the API definition are not used to construct this path. This comment is misleading and could cause confusion or misdirect future security efforts.

File: gateway/api_definition.go [hunk 1]
```diff
@@ -562,11 +562,25 @@ func (a APIDefinitionLoader) replaceVaultSecrets(input *string) error {
...
 	}
 
+	if secret == nil {
+		return fmt.Errorf("vault path does not exist: %s%s; vault references in API definitions will not be resolved", vaultSecretPath, prefixKeys)
+	}
+
+	if secret.Data == nil {
+		return fmt.Errorf("vault path contains no data: %s%s; vault references in API definitions will not be resolved", vaultSecretPath, prefixKeys)
+	}
+
```
Comment (edsonmichaque) [inline gateway/api_definition.go:594]: @chrisanderton the PR looks good! Do you think the above visor recommendation is worth checking? Otherwise LGTM

**Review by edsonmichaque — APPROVED** [review]
LGTM!
@pvormste since you raised some questions, would you give an approval if you are happy?

**Review by pvormste — APPROVED** [review]
Thank you, this looks much better now, great job!
Please note that tests are absolutely required even for such small changes, otherwise we will not be able to catch possible regressions in the future.
